python/lsst/db/mysqlUtils.py

```diff
@@ -0,0 +1,418 @@
+    """
+    Return list of tables in a given database.
+
+    Raises DbException if MySQL error occurs.
+    """
+    cmd = "SELECT TABLE_NAME FROM information_schema.TABLES "
+    cmd += "WHERE TABLE_SCHEMA='%s'" % _sanitize(dbName)
+    try:
+        rows = conn.execute(cmd)
+        return [x[0] for x in rows]
+    except DBAPIError as e:
+        raise DbException(DbException.DRIVER_ERROR,
+                          str(e.orig.args[0]), e.orig.args[1])
+
+def isView(conn, tableName, dbName=None):
+    """
+    Return True if the table <tableName> is a view, False otherwise.
+
+    @param tableName  Table name.
+    @param dbName     Database name.
+
+    @return boolean   True if the table is a view. False otherwise.
+
+    If <dbName> is not set, the current database name will be used.
+
+    Raises DbException if MySQL error occurs.
+    """
+    dbNameStr = "'%s'" % dbName if dbName is not None else "DATABASE()"
+    try:
+        rows = conn.execute("SELECT table_type FROM information_schema.tables "
+              "WHERE table_schema=%s AND table_name='%s'" % (dbNameStr, tableName))
+        row = rows.first()
+        if not row:
+            return False
+        return row[0] == 'VIEW'
+    except DBAPIError as e:
+        raise DbException(DbException.DRIVER_ERROR,
+                          str(e.orig.args[0]), e.orig.args[1])
+
+def getTableContent(conn, tableName, dbName=None):
+    """
+    Get contents of the table <tableName>.
+
+    @param tableName  Table name.
+    @param dbName     Database name.
+
+    @return string    Contents of the table: a string containing output from
+                      "SELECT *", or "<tableName> is empty".
+
+    Raises DbException if MySQL error occurs.
+    """
+    dbNameStr = "`%s`." % dbName if dbName is not None else ""
+    try:
+        ret = conn.execute("SELECT * FROM %s`%s`" % (dbNameStr, tableName))
+    except DBAPIError as e:
+        raise DbException(DbException.DRIVER_ERROR,
+                          str(e.orig.args[0]), e.orig.args[1])
+
+    s = StringIO.StringIO()
+    s.write(tableName)
+    if len(ret) == 0:
+        s.write(" is empty.\n")
+    else:
+        s.write(':\n')
+    for r in ret:
+        print >> s, "   ", r
+    return s.getvalue()
+
+#### User-related functions ########################################################
+def userExists(conn, userName, hostName):
+    """
+    Return True if user <hostName>@<userName> exists, False otherwise.
+
+    Raises DbException if MySQL error occurs.
+    """
+    try:
+        return conn.execute(
+            "SELECT COUNT(*) FROM mysql.user WHERE user='%s' AND host='%s'" % \
+            (userName, hostName)).first()[0] == 1
+    except DBAPIError as e:
+        raise DbException(DbException.DRIVER_ERROR,
+                          str(e.orig.args[0]), e.orig.args[1])
+
+#### Loading sql script ############################################################
+def loadSqlScript(scriptPath, **kwargs):
```

Comment:
I think we should get rid of this entirely, I don't think we use it anywhere and it's kind of problematic code.